tests/subsys/net/lib/aws_fota/aws_fota_json/src/main.c

```diff
@@ -30,8 +30,51 @@ void test_parse_job_execution(void)
 
 	ret = aws_fota_parse_DescribeJobExecution_rsp(encoded,
 						      sizeof(encoded) - 1,
-						      job_id, hostname,
-						      file_path,
+						      job_id,
+						      hostname, sizeof(hostname),
+						      file_path, sizeof(file_path),
+						      &version_number);
+	TEST_ASSERT_EQUAL(1, ret);
+	TEST_ASSERT_EQUAL_STRING(expected_job_id, job_id);
+	TEST_ASSERT_EQUAL(expected_version_number, version_number);
+	TEST_ASSERT_EQUAL_STRING(expected_hostname, hostname);
+	TEST_ASSERT_EQUAL_STRING(expected_file_path, file_path);
+}
+
+void test_parse_job_execution_single_url(void)
+{
+	int ret;
+	int version_number;
+	int expected_version_number = 1;
+	char expected_job_id[] = "9b5caac6-3e8a-45dd-9273-c1b995762f4a";
+	char expected_hostname[] = "fota-update-bucket.s3.eu-central-1.amazonaws.com";
+	char expected_file_path[] = "update.bin?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential="
+								"AKIAWXEL53DXIU7W72AE%2F20190606%2Feu-central-1%2Fs3%2Faws4_request"
+								"&X-Amz-Date=20190606T081505Z&X-Amz-Expires=604800&X-Amz-Signature="
+								"913e00b97efe5565a901df4ff0b87e4878a406941d711f59d45915035989adcc"
+								"&X-Amz-SignedHeaders=host";
+	char encoded[] = "{\"timestamp\":1559808907,\"execution\":{\"jobId\":"
+					 "\"9b5caac6-3e8a-45dd-9273-c1b995762f4a\",\"status\":\"QUEUED\","
+					 "\"queuedAt\":1559808906,\"lastUpdatedAt\":1559808906,\"versionNumber\":1,"
+					 "\"executionNumber\":1,\"jobDocument\":{\"operation\":\"app_fw_update\","
+					 "\"fwversion\":\"2\",\"size\":181124,\"location\":{\"protocol\":\"https:\","
+					 "\"url\":\"https://fota-update-bucket.s3.eu-central-1.amazonaws.com/"
+					 "update.bin?X-Amz-Algorithm=AWS4-HMAC-SHA256"
+					 "&X-Amz-Credential=AKIAWXEL53DXIU7W72AE%2F20190606%2Feu"
+					 "-central-1%2Fs3%2Faws4_request&X-Amz-Date=20190606T081505Z"
+					 "&X-Amz-Expires=604800"
+					 "&X-Amz-Signature="
+					 "913e00b97efe5565a901df4ff0b87e4878a406941d711f59d45915035989adcc"
+					 "&X-Amz-SignedHeaders=host\"}}}}";
+	char job_id[100];
+	char hostname[100];
```

Comment:
When changing this to contain a too small buffer I get some error that seem related to the Job ID, not hostname. Worth looking into.

*** Booting Zephyr OS build v3.3.99-ncs1-2848-g4579f028e1e6 ***
src/main.c:18:test_parse_job_execution:PASS
src/main.c:80:test_parse_job_execution_single_url:FAIL: Expected '' Was '9b5caac6-3e8a-45dd-9273-c1b995762f4a'
src/main.c:86:test_parse_malformed_job_execution:PASS
src/main.c:104:test_parse_job_execution_missing_host_field:PASS
src/main.c:123:test_parse_job_execution_missing_path_field:PASS
src/main.c:141:test_parse_job_execution_missing_job_id_field:PASS
src/main.c:159:test_parse_job_execution_missing_location_obj:PASS
src/main.c:177:test_update_job_longer_than_max:PASS
src/main.c:194:test_timestamp_only:PASS
src/main.c:212:test_update_job_exec_rsp_minimal:PASWhen changing this to contain a too small buffer I get some error that seem related to the Job ID, not hostname. Worth looking into.

*** Booting Zephyr OS build v3.3.99-ncs1-2848-g4579f028e1e6 ***
src/main.c:18:test_parse_job_execution:PASS
src/main.c:80:test_parse_job_execution_single_url:FAIL: Expected '' Was '9b5caac6-3e8a-45dd-9273-c1b995762f4a'
src/main.c:86:test_parse_malformed_job_execution:PASS
src/main.c:104:test_parse_job_execution_missing_host_field:PASS
src/main.c:123:test_parse_job_execution_missing_path_field:PASS
src/main.c:141:test_parse_job_execution_missing_job_id_field:PASS
src/main.c:159:test_parse_job_execution_missing_location_obj:PASS
src/main.c:177:test_update_job_longer_than_max:PASS
src/main.c:194:test_timestamp_only:PASS
src/main.c:212:test_update_job_exec_rsp_minimal:PAS